bot/exts/evergreen/avatar_modification/avatar_modify.py

```diff
@@ -308,7 +313,57 @@ async def spookyavatar(self, ctx: commands.Context, member: discord.Member = Non
 
             await ctx.send(file=file, embed=embed)
 
+    @avatar_modify.command(name="mosaic", root_aliases=("mosaic",))
+    async def mosaic_command(self, ctx: commands.Context, squares: int = 16) -> None:
+        """Splits your avatar into x squares, randomizes them and stitches them back into a new image!"""
+        async with ctx.typing():
+            member = await self._fetch_member(ctx.author.id)
+            if not member:
+                await ctx.send(f"{Emojis.cross_mark} Could not get member info.")
+                return
+
+            if squares < 1:
+                raise commands.BadArgument("Squares must be a positive number")
+
+            if not math.sqrt(squares).is_integer():
+                raise commands.BadArgument("Squares must be a perfect square")
+
+            if squares > MAX_SQUARES:
+                raise commands.BadArgument(f"Number of squares cannot be higher than {MAX_SQUARES:,}.")
```

Comment:
```suggestion
            if 1 <= squares <= MAX_SQUARES:
                raise commands.BadArgument(f"Squares must be a positive number less than or equal to {MAX_SQUARES}")

            if not math.sqrt(squares).is_integer():
                raise commands.BadArgument("Squares must be a perfect square")
```
The "less than max squares" check should be done before the perfect square check, to avoid computing the root of any unnecessarily large inputs